### PR TITLE
docs: updated snippets and samples for profile and analyze methods

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1364)

### CODE REVIEW

File: google/cloud/spanner/samples/samples.cc [hunk 1]
```diff
@@ -1251,6 +1251,41 @@ void QueryNewColumn(google::cloud::spanner::Client client) {
 }
 //! [END spanner_query_data_with_new_column]
 
+void ProfileQuery(google::cloud::spanner::Client client) {
+  namespace spanner = ::google::cloud::spanner;
+
+  //! [profile-query]
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: A general question that I have also been struggling with:  Is it better for the snippet to include any necessary `using` declarations and `namespace` alias declarations, or to omit them?  That is, is the brevity worth the incompleteness?

File: google/cloud/spanner/samples/samples.cc [hunk 2]
```diff
@@ -1434,6 +1475,41 @@ void DmlStandardUpdate(google::cloud::spanner::Client client) {
...
+        if (!update) return update.status();
+        dml_result = std::move(update);
+        return spanner::Mutations{};
+      });
+  if (!commit_result) {
+    throw std::runtime_error(commit_result.status().message());
+  }
+
+  // Stats only available after statement has been executed.
+  if (dml_result) {
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: Can this test ever fail?

File: google/cloud/spanner/samples/samples.cc [hunk 3]
```diff
@@ -2090,11 +2165,26 @@ void PartitionRead(google::cloud::spanner::Client client) {
...
+  // `ExecutionPlan` can be partitioned.
+  google::cloud::StatusOr<spanner::ExecutionPlan> plan = client.AnalyzeSql(
+      spanner::MakeReadOnlyTransaction(),
+      spanner::SqlStatement(
+          "SELECT SingerId, FirstName, LastName FROM Singers"));
+  if (!plan) throw std::runtime_error(plan.status().message());
+  if (plan->plan_nodes(plan->plan_nodes_size() - 1).kind() !=
+          google::spanner::v1::PlanNode::RELATIONAL &&
+      plan->plan_nodes(plan->plan_nodes_size() - 1).display_name() !=
+          "Distributed Union")
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: Nit: Braces might be nice on this `if`.

Comment (scotthart) [reply google/cloud/spanner/samples/samples.cc]: Done.

**Pull request merged**